benchmarks/REPS_CREPS_gradients/creps_benchmark.py

```diff
@@ -0,0 +1,162 @@
+    "C-REPS-NUM": CREPSOptimizer,
+    "C-REPS-AN": CREPSOptimizer
+}
+seeds = list(range(20))
+
+n_samples_per_update = 50
+additional_ctor_args = {
+    "C-REPS-NUM": {
+        "train_freq": n_samples_per_update,
+        "epsilon": 1.0,
+        "min_eta": 1e-10,  # 1e-20 in the original code
+        "approx_grad": True
+    },
+    "C-REPS-AN": {
+        "train_freq": n_samples_per_update,
+        "epsilon": 1.0,
+        "min_eta": 1e-10  # 1e-20 in the original code
+    }
+}
+n_params = 20
+context_dims_per_objective = {
+    "sphere": 2,
+}
+n_episodes_per_objective = {
+    "sphere": 15 * n_samples_per_update,
+}
+
+def benchmark():
+    """Run benchmarks for all configurations of objective and algorithm."""
+    results = dict(
+        (objective_name,
+         dict((algorithm_name, [])
+              for algorithm_name in algorithms.keys()))
+        for objective_name in objective_functions.keys()
+    )
+    for objective_name in objective_functions.keys():
+        for algorithm_name in algorithms.keys():
+            start_time = time.time()
+            feedbacks = Parallel(n_jobs=n_jobs, verbose=10)(
+                delayed(optimize)(objective_name, algorithm_name, seed)
+                for run_idx, seed in enumerate(seeds))
+            results[objective_name][algorithm_name] = feedbacks
+            completion_time = time.time() - start_time
+            print 'Algorithm', algorithm_name, 'and objective function ', \
+                objective_name, 'took', completion_time, 'seconds to complete'
+    return results
+
+def optimize(objective_name, algorithm_name, seed):
+    """Perform one benchmark run."""
+    n_context_dims = context_dims_per_objective[objective_name]
+    n_episodes = n_episodes_per_objective[objective_name]
+    random_state = np.random.RandomState(seed)
+    # contexts are sampled uniformly from 1 <= s <= 2 (here: < 2)
+    contexts = random_state.rand(n_episodes, n_context_dims) + 1.0
+    obj = objective_functions[objective_name](random_state, n_params, n_context_dims)
+    initial_params = random_state.randn(n_params)
+    opt = algorithms[algorithm_name](
+        initial_params=initial_params,
+        covariance=np.eye(n_params),
+        variance=1.0,
+        n_samples_per_update=n_samples_per_update,
+        context_features="affine",
+        random_state=random_state,
+        gamma=1e-10,
+        **additional_ctor_args[algorithm_name]
+    )
+    opt.init(n_params, n_context_dims)
+
+    feedbacks = np.empty(n_episodes)
+    params = np.empty(n_params)
+    for episode_idx in range(n_episodes):
+        opt.set_context(contexts[episode_idx])
+        opt.get_next_parameters(params)
+        feedbacks[episode_idx] = obj.feedback(
+            params, contexts[episode_idx])
+        opt.set_evaluation_feedback(feedbacks[episode_idx])
+
+    return feedbacks
+
+def show_results(results):
+    """Display results."""
+    for objective_name, objective_results in results.items():
```

Comment:
You cannot recognize any difference between both curves. Maybe there is a better way, for example, using a dashed and a solid line.

Reply:
I have used a dashed line so that both lines can be seen, but there is no difference between both curves as the solutions found are the same.

Reply:
Yes, looks better now!